### ISSUE TITLE
Writes [object Object] before index.html </head>

### ISSUE THREAD

**Issue by jason21000000** [davewasmer/ember-cli-favicon#2]
I installed with "ember install ember-cli-favicon", I created a 256x256 png at /public/favicon.png. After rebuilding / restarting the server, I see [object Object] right before the < / head > in my index.html

I am using Ember 1.13.6

**Comment by SeyZ** [comment]
Same behavior here. :+1:

**Comment by turboMaCk** [comment]
+1

**Comment by turboMaCk** [comment]
this is fix: https://github.com/davewasmer/ember-cli-favicon/pull/3

**Comment by davewasmer** [comment]
Sorry for the delayed reply here! I'm unable to reproduce this particular issue. Any chance anyone could post a repo isolating the issue?

@turboMaCk - I'm not sure how your PR fixes things? It seems like you moved where the default value is used, and changed the default from `null` to `''`. But the symptoms described seem to indicate that the HTML callback is returning an object rather than an HTML string, so I'm not sure I understand how your fix solves that.

**Comment by turboMaCk** [comment]
@davewasmer I thought that issue is related with this call `toString(null) => "[object Window]"` But it looks like I was wrong. But I know how to reproduce this issue. It happens after first build => when cache is created. In every other build (which use cached images) everything is OK.

**Comment by Blackening999** [comment]
Hey guys!

 I have the same issue here. Please update here when it fixed

**Comment by turboMaCk** [comment]
I think it will be best to write some tests at first.

**Comment by davewasmer** [comment]
Yea, I'd like to get some tests for this (and broccoli-favicon) up, as well as convert to ES6. PR's welcome :wink:

**Comment by davewasmer** [comment]
So I did some digging, turns out there's a dependency deep down here [1] which is potentially causing occasional failures. I filed an issue and will update if/when it's fixed.

However, if I'm correct, I think you guys should be seeing some error output during the build process, specifically an `invalid signature` error. Has anyone seen errors like that? And is the `[object Object]` appearing every time, or just occasionally?

If not, can anyone reproduce in an isolated repo? I still can't reproduce locally ...

[1] ember-cli-favicon -> broccoli-favicon -> favicons -> unzip

**Comment by davewasmer** [comment]
Hey all - I think I found the actual source of the issue. The `invalid signature` bit I mentioned above is probably unrelated.

Looks like I didn't handle errors from the favicon API very well. I got these symptoms when I accidentally exported a favicon.png with non-square dimensions. Could anyone here check to see if their source file was not square?

I'll be publishing a release we better error handling, hopefully that will help.
